src/test/java/com/github/wovnio/wovnjava/ApiTest.java

```diff
@@ -66,18 +87,32 @@ private String translate(HttpServletRequest request, Settings settings, String h
         return buffer.toByteArray();
     }
 
-    private HttpURLConnection mockHttpURLConnection( int sendLength, int code, String encoding, byte[] response) throws IOException, ProtocolException {
-        InputStream responseStream = new ByteArrayInputStream(response);
+    private HttpURLConnection mockHttpURLConnection(ByteArrayOutputStream requestStream, ByteArrayInputStream responseStream, int code, String encoding) throws IOException, ProtocolException {
         HttpURLConnection mock = EasyMock.createMock(HttpURLConnection.class);
         mock.setDoOutput(true);
         mock.setRequestProperty(EasyMock.anyString(), EasyMock.anyString());
         EasyMock.expectLastCall().atLeastOnce();
         mock.setRequestMethod("POST");
         EasyMock.expect(mock.getResponseCode()).andReturn(code);
         EasyMock.expect(mock.getContentEncoding()).andReturn(encoding);
-        EasyMock.expect(mock.getOutputStream()).andReturn(new ByteArrayOutputStream());
+        EasyMock.expect(mock.getOutputStream()).andReturn(requestStream);
         EasyMock.expect(mock.getInputStream()).andReturn(responseStream);
         EasyMock.replay(mock);
         return mock;
     }
+
+    private static String decompress(byte[] compressed) throws IOException {
+        ByteArrayInputStream bis = new ByteArrayInputStream(compressed);
+        GZIPInputStream gis = new GZIPInputStream(bis);
+        BufferedReader br = new BufferedReader(new InputStreamReader(gis, "UTF-8"));
+        StringBuilder sb = new StringBuilder();
+        String line;
+        while((line = br.readLine()) != null) {
+            sb.append(line);
+        }
+        br.close();
```

Comment:
Don't we have to care about exceptional safety?

Reply:
I think that it's okay because this is test.

Reply:
Thanks, I'll close the streams in a `finally` clause.